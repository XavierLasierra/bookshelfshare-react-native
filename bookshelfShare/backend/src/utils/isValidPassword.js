const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
