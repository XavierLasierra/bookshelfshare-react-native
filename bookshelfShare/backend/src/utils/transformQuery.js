function transformQuery(query) {
  const entries = Object.entries(query);
  return entries.reduce((acc, [property, filter]) => [...acc, { [property]: { $regex: filter, $options: 'i' } }],
    []);
}

module.exports = { transformQuery };
