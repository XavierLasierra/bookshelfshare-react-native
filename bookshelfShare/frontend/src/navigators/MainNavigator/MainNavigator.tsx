import React from 'react';
import {useSelector} from 'react-redux';
import {createSharedElementStackNavigator} from 'react-navigation-shared-element';
import {CardStyleInterpolators} from '@react-navigation/stack';

import InitialLoading from '../../components/InitialLoading/InitialLoading';
import Login from '../../components/Login/Login';
import Register from '../../components/Register/Register';
import PagesNavigator from '../PagesNavigator/PagesNavigator';

export default function MainNavigator() {
  const Stack = createSharedElementStackNavigator();

  const {isAuthenticated} = useSelector((store: any) => store.loggedUser);

  return (
    <Stack.Navigator
      initialRouteName="InitialLoading"
      screenOptions={{
        headerShown: false,
        cardShadowEnabled: false,
        cardOverlayEnabled: false,
        cardStyleInterpolator: CardStyleInterpolators.forFadeFromBottomAndroid,
      }}>
      {!isAuthenticated ? (
        <>
          <Stack.Screen name="InitialLoading" component={InitialLoading} />
          <Stack.Screen
            name="Login"
            component={Login}
            sharedElements={() => ['mainIcon']}
          />
          <Stack.Screen
            name="Register"
            component={Register}
            sharedElements={() => ['mainIcon']}
            options={{
              cardStyleInterpolator: CardStyleInterpolators.forFadeFromCenter,
            }}
          />
        </>
      ) : (
        <Stack.Screen name="PagesNavigator" component={PagesNavigator} />
      )}
    </Stack.Navigator>
  );
}
