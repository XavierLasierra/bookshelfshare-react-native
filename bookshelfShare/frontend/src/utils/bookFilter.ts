export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
