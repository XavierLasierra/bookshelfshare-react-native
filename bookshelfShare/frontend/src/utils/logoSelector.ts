import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
