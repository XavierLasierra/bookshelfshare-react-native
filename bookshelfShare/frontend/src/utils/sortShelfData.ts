export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
