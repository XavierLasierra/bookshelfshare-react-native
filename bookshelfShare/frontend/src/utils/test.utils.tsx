import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
