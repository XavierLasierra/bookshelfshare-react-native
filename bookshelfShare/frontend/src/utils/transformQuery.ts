interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
