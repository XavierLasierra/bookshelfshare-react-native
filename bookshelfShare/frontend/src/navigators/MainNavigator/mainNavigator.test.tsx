import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { MenuProvider } from 'react-native-popup-menu';
import MainNavigator from './MainNavigator';
import { render } from '../../utils/test.utils';

jest.mock('../../redux/actions/loggedUser.creator', () => ({
  automaticLogin: () => ({ type: '' })
}));

jest.useFakeTimers();

describe('Given a MainNavigator component', () => {
  describe('When it is rendered', () => {
    describe('And the user is not authenticated', () => {
      test('Then should match the snapshot', () => {
        const initialState = { loggedUser: { isAuthenticated: false, needsLogin: false } };
        const screen = render(
          <NavigationContainer>
            <MainNavigator />
          </NavigationContainer>,
          initialState
        );
        expect(screen).toMatchSnapshot();
      });
    });
    describe('And the user is authenticated', () => {
      test('Then should match the snapshot', () => {
        const initialState = { loggedUser: { isAuthenticated: true } };
        const screen = render(
          <NavigationContainer>
            <MenuProvider>
              <MainNavigator />
            </MenuProvider>
          </NavigationContainer>,
          initialState
        );

        expect(screen).toMatchSnapshot();
      });
    });
  });
});
