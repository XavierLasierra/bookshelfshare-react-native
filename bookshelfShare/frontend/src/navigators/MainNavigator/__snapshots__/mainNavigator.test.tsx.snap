// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MainNavigator component When it is rendered And the user is authenticated Then should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNCSafeAreaProvider
        onInsetsChange={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              />
              <View
                accessibilityElementsHidden={false}
                closing={false}
                gestureVelocityImpact={0.3}
                importantForAccessibility="auto"
                onClose={[Function]}
                onGestureBegin={[Function]}
                onGestureCanceled={[Function]}
                onGestureEnd={[Function]}
                onOpen={[Function]}
                onTransition={[Function]}
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "overflow": undefined,
                    },
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                  ]
                }
                transitionSpec={
                  Object {
                    "close": Object {
                      "animation": "spring",
                      "config": Object {
                        "damping": 500,
                        "mass": 3,
                        "overshootClamping": true,
                        "restDisplacementThreshold": 10,
                        "restSpeedThreshold": 10,
                        "stiffness": 1000,
                      },
                    },
                    "open": Object {
                      "animation": "spring",
                      "config": Object {
                        "damping": 500,
                        "mass": 3,
                        "overshootClamping": true,
                        "restDisplacementThreshold": 10,
                        "restSpeedThreshold": 10,
                        "stiffness": 1000,
                      },
                    },
                  }
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  pointerEvents="box-none"
                  style={
                    Object {
                      "flex": 1,
                      "marginTop": 0,
                    }
                  }
                >
                  <View
                    enableTrackpadTwoFingerGesture={true}
                    enabled={false}
                    hitSlop={
                      Object {
                        "right": -700,
                      }
                    }
                    maxDeltaY={20}
                    minOffsetX={5}
                    onHandlerStateChange={[Function]}
                  >
                    <View
                      needsOffscreenAlphaCompositing={true}
                      style={
                        Object {
                          "flex": 1,
                          "opacity": 1,
                          "transform": Array [
                            Object {
                              "translateY": 0,
                            },
                          ],
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "flex": 1,
                              "overflow": "hidden",
                            },
                            Array [
                              Object {
                                "backgroundColor": "rgb(242, 242, 242)",
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "column-reverse",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "flex": 1,
                                      "overflow": "hidden",
                                    }
                                  }
                                >
                                  <View
                                    pointerEvents="auto"
                                    style={
                                      Array [
                                        Object {
                                          "flex": 1,
                                          "overflow": "hidden",
                                        },
                                        Array [
                                          Object {
                                            "bottom": 0,
                                            "left": 0,
                                            "position": "absolute",
                                            "right": 0,
                                            "top": 0,
                                          },
                                          Object {
                                            "zIndex": 0,
                                          },
                                        ],
                                      ]
                                    }
                                  >
                                    <View
                                      collapsable={false}
                                      pointerEvents="auto"
                                      removeClippedSubviews={false}
                                      style={
                                        Object {
                                          "flex": 1,
                                        }
                                      }
                                    >
                                      <View
                                        accessibilityElementsHidden={false}
                                        importantForAccessibility="auto"
                                        style={
                                          Array [
                                            Object {
                                              "backgroundColor": "rgb(242, 242, 242)",
                                              "flex": 1,
                                            },
                                            Array [
                                              Object {
                                                "flex": 1,
                                                "flexDirection": "column-reverse",
                                              },
                                              undefined,
                                            ],
                                          ]
                                        }
                                      >
                                        <View
                                          style={
                                            Object {
                                              "flex": 1,
                                            }
                                          }
                                        >
                                          <RCTSafeAreaView
                                            emulateUnlessSupported={true}
                                          >
                                            <View
                                              style={
                                                Object {
                                                  "alignItems": "flex-end",
                                                  "backgroundColor": "#F5F5F5",
                                                }
                                              }
                                            >
                                              <View
                                                style={
                                                  Object {
                                                    "alignItems": "center",
                                                    "backgroundColor": "#D9D8C4",
                                                    "borderBottomLeftRadius": 25,
                                                    "justifyContent": "center",
                                                    "padding": 10,
                                                    "width": "100%",
                                                  }
                                                }
                                              >
                                                <IconMock
                                                  height={45}
                                                  width={45}
                                                />
                                              </View>
                                              <View
                                                style={
                                                  Object {
                                                    "height": 0,
                                                  }
                                                }
                                              >
                                                <View
                                                  style={
                                                    Object {
                                                      "backgroundColor": "#D9D8C4",
                                                      "height": 20,
                                                      "width": 20,
                                                    }
                                                  }
                                                >
                                                  <View
                                                    style={
                                                      Object {
                                                        "backgroundColor": "#F5F5F5",
                                                        "borderTopRightRadius": 25,
                                                        "height": 20,
                                                        "width": 20,
                                                      }
                                                    }
                                                  />
                                                </View>
                                              </View>
                                            </View>
                                          </RCTSafeAreaView>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "backgroundColor": "#FFFFFF",
                                      "flexDirection": "row",
                                      "height": 60,
                                      "justifyContent": "space-around",
                                    }
                                  }
                                >
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#F5F5F5",
                                        "borderRadius": 50,
                                        "height": 57,
                                        "justifyContent": "center",
                                        "opacity": 1,
                                        "width": 57,
                                      }
                                    }
                                    testID="homeButton"
                                  >
                                    <IconMock
                                      height={30}
                                      width={30}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "color": "#3E3C37",
                                          "fontSize": 12,
                                        }
                                      }
                                    >
                                      home
                                    </Text>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "height": 57,
                                        "justifyContent": "center",
                                        "opacity": 1,
                                        "width": 57,
                                      }
                                    }
                                    testID="followingButton"
                                  >
                                    <IconMock
                                      height={30}
                                      width={30}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "color": "#3E3C37",
                                          "fontSize": 12,
                                        }
                                      }
                                    >
                                      following
                                    </Text>
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#FFFFFF",
                                        "borderRadius": 50,
                                        "elevation": 2,
                                        "height": 60,
                                        "justifyContent": "center",
                                        "position": "relative",
                                        "top": -10,
                                        "width": 60,
                                      }
                                    }
                                  >
                                    <View
                                      collapsable={false}
                                    >
                                      <View
                                        accessible={true}
                                        focusable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        testID="openMenuButton"
                                      >
                                        <View
                                          ctx={
                                            Object {
                                              "menuActions": Object {
                                                "_getOpenedMenu": [Function],
                                                "_notify": [Function],
                                                "closeMenu": [Function],
                                                "isMenuOpen": [Function],
                                                "openMenu": [Function],
                                                "toggleMenu": [Function],
                                              },
                                              "menuRegistry": Object {
                                                "getAll": [Function],
                                                "getMenu": [Function],
                                                "setOptionsCustomStyles": [Function],
                                                "subscribe": [Function],
                                                "unsubscribe": [Function],
                                                "updateLayoutInfo": [Function],
                                              },
                                            }
                                          }
                                          style={
                                            Array [
                                              undefined,
                                              undefined,
                                            ]
                                          }
                                        >
                                          <Text
                                            style={
                                              Object {
                                                "color": "#3E3C37",
                                                "fontSize": 30,
                                                "height": 60,
                                                "lineHeight": 60,
                                                "textAlign": "center",
                                                "width": 60,
                                              }
                                            }
                                          >
                                            +
                                          </Text>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "height": 57,
                                        "justifyContent": "center",
                                        "opacity": 1,
                                        "width": 57,
                                      }
                                    }
                                    testID="shelfButton"
                                  >
                                    <IconMock
                                      height={30}
                                      width={30}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "color": "#3E3C37",
                                          "fontSize": 12,
                                        }
                                      }
                                    >
                                      shelf
                                    </Text>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "height": 57,
                                        "justifyContent": "center",
                                        "opacity": 1,
                                        "width": 57,
                                      }
                                    }
                                    testID="profileButton"
                                  >
                                    <IconMock
                                      height={30}
                                      width={30}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "color": "#3E3C37",
                                          "fontSize": 12,
                                        }
                                      }
                                    >
                                      profile
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              }
            }
          />
        </View>
      </RNCSafeAreaProvider>
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Given a MainNavigator component When it is rendered And the user is not authenticated Then should match the snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                Array [
                  Object {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                Object {
                  "close": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                pointerEvents="box-none"
                style={
                  Object {
                    "flex": 1,
                    "marginTop": 0,
                  }
                }
              >
                <View
                  enableTrackpadTwoFingerGesture={true}
                  enabled={false}
                  hitSlop={
                    Object {
                      "right": -700,
                    }
                  }
                  maxDeltaY={20}
                  minOffsetX={5}
                  onHandlerStateChange={[Function]}
                >
                  <View
                    needsOffscreenAlphaCompositing={true}
                    style={
                      Object {
                        "flex": 1,
                        "opacity": 1,
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "backgroundColor": "rgb(242, 242, 242)",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "column-reverse",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <RCTSafeAreaView
                              emulateUnlessSupported={true}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#D9D8C4",
                                  "display": "flex",
                                  "height": "100%",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "height": 30,
                                  }
                                }
                              >
                                <View
                                  collapsable={false}
                                >
                                  <IconMock
                                    height={60}
                                    width={60}
                                  />
                                </View>
                              </View>
                            </RCTSafeAreaView>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 44,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 1,
            }
          }
        />
      </View>
    </RNCSafeAreaProvider>
  </View>,
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />,
]
`;
